ddpui/ddpairbyte/airbytehelpers.py

```diff
@@ -1079,14 +1224,39 @@ def schedule_update_connection_schema(
     if not dataflow_orgtask:
         raise HttpError(400, "no dataflow mapped")
 
-    locks: list[TaskLock] = prefect_service.lock_tasks_for_deployment(
-        dataflow_orgtask.dataflow.deployment_id,
-        orguser,
-        dataflow_orgtasks=[dataflow_orgtask],
-    )
+    # check if the connection is "large" for scheduling
+    connection_meta = ConnectionMeta.objects.filter(connection_id=connection_id).first()
+    is_connection_large_enough = connection_meta and connection_meta.schedule_large_jobs
+
+    logger.info("connection is large enough: %s", is_connection_large_enough)
 
+    locks: list[TaskLock] = []
+    schedule_at = None
+    job: ConnectionJob = None
+    if not is_connection_large_enough:
+        locks = prefect_service.lock_tasks_for_deployment(
+            dataflow_orgtask.dataflow.deployment_id,
+            orguser,
+            dataflow_orgtasks=[dataflow_orgtask],
+        )
+    else:
+        schedule_at = get_schedule_time_for_large_jobs()
+        # if there is a flow run scheduled , delete it
+        job = ConnectionJob.objects.filter(
+            connection_id=connection_id, job_type=UPDATE_SCHEMA
+        ).first()
+        if job:
+            try:
+                prefect_service.delete_flow_run(job.flow_run_id)
```

Comment:
set `job.flow_run_id = None` and save, just in case there's a failure before the update below